lib/mix/lib/mix/tasks/xref.ex

```diff
@@ -34,12 +36,13 @@ defmodule Mix.Tasks.Xref do
   """
 
   @switches [compile: :boolean, deps_check: :boolean, archives_check: :boolean,
-             warnings: :boolean, unreachable: :boolean, elixir_version_check: :boolean]
+             warnings: :boolean, unreachable: :boolean, elixir_version_check: :boolean,
+             callers: :string]
 
   @doc """
   Runs this task.
   """
-  @spec run(OptionParser.argv) :: :ok | :error
+  @spec run(OptionParser.argv) :: :ok | :error | [{Path.t, [{atom, atom, non_neg_integer, [pos_integer]}]}]
```

Comment:
Do we really need to return the paths from the task? I would keep it as `:ok | :error`. If we have to expose those, I would rather have a proper API than do it through Mix.Task.run. :)